### PR TITLE
feat: add generics annotations to dataSources

### PR DESCRIPTION
**Pull request by wzs**

### CODE REVIEW

File: TWCommonLib/TWCommonLib/TWCoreDataTableViewDataSource.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
 typedef NSString* (^CellReuseMappingBlock)(NSIndexPath *);
 
 
-@interface TWCoreDataTableViewDataSource : NSObject <UITableViewDataSource>
+@interface TWCoreDataTableViewDataSource<__covariant T:NSManagedObjectModel *> : NSObject <UITableViewDataSource>
```
Comment (wyszo) [inline TWCommonLib/TWCommonLib/TWCoreDataTableViewDataSource.h]: Why NSManagedObjectModel? I think you meant NSManagedObject in here?

**Comment by wyszo** [comment]
Generics FTW! :D 
:dancer: :dancer: :dancer:

File: TWCommonLib/TWCommonLib/TWCoreDataTableViewDataSource.h (hunk 1 above)
Comment (wzs) [inline TWCommonLib/TWCommonLib/TWCoreDataTableViewDataSource.h]: Right, stupid mistake

Comment (wyszo) [inline TWCommonLib/TWCommonLib/TWCoreDataTableViewDataSource.h]: Sweet! Thanks for the contribution!

**Pull request merged**